Data/ByteString/Char8.hs

```diff
@@ -963,3 +963,35 @@ readInteger as
 
           combine2 b (n:m:ns) = let t = m*b + n in t `seq` (t : combine2 b ns)
           combine2 _ ns       = ns
+
+------------------------------------------------------------------------
+-- For non-binary text processing:
+
+-- | Read an entire file strictly into a 'ByteString'.  This is far more
+-- efficient than reading the characters into a 'String' and then using
+-- 'pack'.  It also may be more efficient than opening the file and
+-- reading it using hGet.
+readFile :: FilePath -> IO ByteString
+readFile f = withBinaryFile f ReadMode
+    (\h -> hFileSize h >>= hGet h . fromIntegral)
+
+modifyFile :: IOMode -> FilePath -> ByteString -> IO ()
+modifyFile mode f txt = withBinaryFile f mode (`hPut` txt)
+
+-- | Write a 'ByteString' to a file.
+writeFile :: FilePath -> ByteString -> IO ()
+writeFile = modifyFile WriteMode
+
+-- | Append a 'ByteString' to a file.
+appendFile :: FilePath -> ByteString -> IO ()
+appendFile = modifyFile AppendMode
```

Comment:
Wait, the only ones that are deprecated in Data.ByteString are `putStrLn` and `hPutStrLn`, and so those are the two that ought to be implemented here in the .Char8 module, all the others above are ok to re-export from Data.ByteString and not implement here right?